server/src/main/java/com/cloud/template/HypervisorTemplateAdapter.java

```diff
@@ -665,8 +669,13 @@ public boolean delete(TemplateProfile profile) {
             // Remove template details
             templateDetailsDao.removeDetails(template.getId());
 
-            // Remove deploy-as-is details (if any)
-            templateDeployAsIsDetailsDao.removeDetails(template.getId());
+            // Remove deploy-as-is details (if any and if there are no VMs using it)
```

Comment:
letś make this a javadoc of a new method

Reply:
Added a new method and java doc